src/expr/src/relation/func.rs

```diff
@@ -978,6 +1084,28 @@ impl AggregateFunc {
                     custom_id: None,
                 }
             }
+            AggregateFunc::FirstValue { .. } => {
+                // The input type for FirstValue is ((OriginalRow, EncodedArgs), OrderByExprs...)
+                let fields = input_type.scalar_type.unwrap_record_element_type();
+                let original_row_type = fields[0].unwrap_record_element_type()[0]
+                    .clone()
+                    .nullable(false);
```

Comment:
Can this indeed not be null? I'm not 100% clear on what's going on here, so I could be missing something.

Reply:
It's guaranteed to be not null by construction because it's [the record created for the original row](https://github.com/MaterializeInc/materialize/blob/main/src/sql/src/plan/lowering.rs#L1118-L1138). The record can just hold a row fulls of nulls, but itself should always be not null.